src/Umbraco.Core/Migrations/Upgrade/V_8_0_0/VariantsMigration.cs

```diff
@@ -90,6 +100,13 @@ private void MigratePropertyData()
             Rename.Table(PreTables.PropertyData).To(Constants.DatabaseSchema.Tables.PropertyData).Do();
         }
 
+        private void CreatePropertyDataIndexes()
+        {
+            var tableDefinition = Persistence.DatabaseModelDefinitions.DefinitionFactory.GetTableDefinition(typeof(PropertyDataDto), SqlSyntax);
+            Execute.Sql(SqlSyntax.FormatPrimaryKey(tableDefinition)).Do();
+            Execute.Sql("CREATE UNIQUE NONCLUSTERED INDEX[IX_umbracoPropertyData_VersionId] ON [umbracoPropertyData]([versionId],[propertyTypeId],[languageId],[segment])").Do();
```

Comment:
I should have made it clear here that this is a temporary index for the benefit of the remaining migrations, since we've got no indexes at this stage. The index is deleted in `CreateKeysAndIndexes.cs` before we create the indexes defined by `PropertyDataDto`.

If memory serves, I ended up with this particular index for the "set 'edited' to true whenever a 'non-published' property data is != a published one" step in `VariantsMigration`, and it covers what's needed for all the `Database.Update(propertyDataDto)` calls too.